backend/src/v2/driver/driver.go

```diff
@@ -480,6 +480,28 @@ func extendPodSpecPatch(
 		podSpec.NodeSelector = kubernetesExecutorConfig.GetNodeSelector().GetLabels()
 	}
 
+	if tolerations := kubernetesExecutorConfig.GetTolerations(); tolerations != nil {
+		var k8sTolerations []k8score.Toleration
+
+		glog.Infof("Tolerations passed: %+v", tolerations)
+
+		for _, toleration := range tolerations {
+			if toleration != nil {
+				k8sToleration := k8score.Toleration{
```

Comment:
I remembered we discussed the need to validate the correctness of this. I would move this out into a function that return a `(k8score.Toleration, error)` to keep the loop simple, and at each iteration either append to the array or return an error and fail early

Reply:
See https://github.com/kubeflow/pipelines/blob/002bfe96903ec1766151a9d6dada7f38303be481/backend/src/v2/driver/driver.go#L1259

Reply:
@connor-mccarthy curious to hear your thoughts on this. Should we add additional validation to the driver logic or let the K8s API handle validation? The main advantage of deferring validation to the K8s API is significantly reduced overhead IMO. Could go either way though.

Reply:
@edmondop we should be able to just pull in https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L3900 and call it in the driver logic! No duplication necessary. Shout out to @DWSR for the excellent suggestion!

Reply:
So `validationTolerations` is in core whereas the podspec declared by Kubeflow uses core/v1. I think validating in the driver doesn't really accomplish anything. The failure is still asynchronous (the run submission will still succeed) and will require parsing the workflow failure to debug. I suggest we defer validation to the K8s API.